hca-to-scea/hca2scea.py

```diff
@@ -585,55 +585,40 @@ def main():
         "Smart-seq": "smart-seq"
     }
 
-    list_xlsx_dict = split_dataset.split_metadata_by_technology(xlsx_dict_tmp2,technology_dict)
+    check_experimental_design.check_technology_eligibility(xlsx_dict, technology_dict)
 
-    if len(list_xlsx_dict) > 1:
-        accession_number_idx = [i for i in range(0,len(list_xlsx_dict))]
-        accession_number_list = [int(args.accession_number) + i for i in accession_number_idx]
+    accession_number = args.accession_number
 
-    else:
-        accession_number_idx = [0]
-        accession_number_list = [args.accession_number]
-
-    for i in range(0,len(list_xlsx_dict)):
-
-        xlsx_dict = list_xlsx_dict[i]
-        accession_number = accession_number_list[i]
-
-        related_scea_accessions_idx = copy.deepcopy(accession_number_idx)
-        related_scea_accessions_idx.remove(i)
-        related_scea_accessions = [accession_number_list[i] for i in related_scea_accessions_idx]
+    '''Run checks to see whether the experimental design is compatibile and save the experimental
+    design type in a variable for later.'''
 
-        '''Run checks to see whether the experimental design is compatibile and save the experimental
-        design type in a variable for later.'''
+    ''' Get the experimental design '''
+    experimental_design = check_experimental_design.get_experimental_design(xlsx_dict)
 
-        ''' Get the experimental design '''
-        experimental_design = check_experimental_design.get_experimental_design(xlsx_dict)
-
-        ''' Check if samples are pooled '''
-        pooled_samples = check_experimental_design.check_for_pooled_samples(xlsx_dict)
-        if pooled_samples:
-            print("The hca-to-scea tool does not support pooled donors or pooled samples."
-                "The dataset should be curated manually.")
-            sys.exit()
+    ''' Check if samples are pooled '''
+    pooled_samples = check_experimental_design.check_for_pooled_samples(xlsx_dict)
+    if pooled_samples:
+        print("The hca-to-scea tool does not support pooled donors or pooled samples."
+            "The dataset should be curated manually.")
+        sys.exit()
```

Comment:
Using sys.exit makes it diffiult to integrate your code into something else at a later stage, e.g. running it from the UI.
A better way is to raise an exception, and catch it higher up in the call chain.

Reply:
This code snippet has actually been removed from the code in a separate branch which checks for pooled samples. In that branch, an exception is raised and it occurs as part of the experimental design checks before the main script functionality.